### PR TITLE
add support for dashboard- and chart-level workflow links

### PR DESCRIPTION
**Pull request by MisterSquishy**
e.g.
```HCL
resource "lightstep_dashboard" "test_workflow_links" {
	...
	workflow_link {
		name = "test dashboard link"
		url  = "https://test.com"
	}

	chart {
		...
		workflow_link {
			name = "test chart link"
			url  = "https://testchart.com"
		}
	}
}
```

### CODE REVIEW

File: lightstep/resource_metric_dashboard.go [hunk 1]
```diff
@@ -471,13 +505,17 @@ func getUnifiedDashboardAttributesFromResource(d *schema.ResourceData) (*client.
 	eventQueriesSet := d.Get("event_query_ids").(*schema.Set)
 	eventQueries := buildStringSlice(eventQueriesSet.List())
 
+	workflowLinksList := d.Get("workflow_link").([]any)
```
[the remark this answers is not in the document]
Comment (ahiseley) [reply lightstep/resource_metric_dashboard.go:508]: type casting with an unexpected shape?

[the remark this answers is not in the document]
Comment (MisterSquishy) [reply lightstep/resource_metric_dashboard.go:508]: yeah, that [d.Get](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/resource_data.go#L57-L66) function is getting the list of workflow links from the terraform state -- but all we told terraform is the [schema](https://github.com/lightstep/terraform-provider-lightstep/pull/232/files#diff-791633a35ea53da274971fb770a1a04b734db8bfb25cb9a661ca2a88a574211fR108-R117), so it doesnt have a strongly-typed representation of the data

im honestly not totally sure why we traditionally cast to `[]any` first, and then do a subsequent cast (down [here](https://github.com/lightstep/terraform-provider-lightstep/pull/232/files#diff-791633a35ea53da274971fb770a1a04b734db8bfb25cb9a661ca2a88a574211fR736)) to get the elements of the array as maps. i think maybe i dont like that pattern and im gonna break it here?? let's see how this goes...

[the remark this answers is not in the document]
Comment (ahiseley) [reply lightstep/resource_metric_dashboard.go:508]: +1 all for seeing how it goes in the face of uncertainty 😆

[the remark this answers is not in the document]
Comment (MisterSquishy) [reply lightstep/resource_metric_dashboard.go:508]: it did not go great 😂 
```
panic: interface conversion: interface {} is []interface {}, not []map[string]string
```
i guess go makes you cast in two steps??

[the remark this answers is not in the document]
Comment (ahiseley) [reply lightstep/resource_metric_dashboard.go:508]: https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExYnVwZmUzcnZpcnN6cTl0dnJqaXhtcDY3aHZrbDZyaHE4eHA5d21kayZlcD12MV9naWZzX3NlYXJjaCZjdD1n/FzXa3cs6ux2so/giphy.gif

[the remark this answers is not in the document]
Comment (MisterSquishy) [reply lightstep/resource_metric_dashboard.go:508]: even just this doesnt work -- i guess generic arrays and maps omit type info, so their elements must be cast individually 😞 
CleanShot 2024-08-08 at 14 40 15@2x

**Pull request merged**